juju/model.py

```diff
@@ -2923,7 +2927,7 @@ async def wait_for_idle(
         status=None,
         wait_for_at_least_units=None,
         wait_for_exact_units=None,
-    ):
+    ) -> None:
```

Comment:
I guess it might be a lot of work to annotate all the other arguments if they're unusually typed?

I'd be in favour of annotating any that are easy to annotate tho, e.g. if `raise_on_error/blocked` and `wait_for_active` are `bool`, if `timeout`, `idle_period`, `check_freq` are `float` (or `int`?), and I'm guessing that `wait_for_..._units` are `int | None`. Is `status` a `str | None`?

Reply:
There will be a separate PR for .wait_for_idle()

For now I'm sprinkled a few type hints, mostly because I needed them somewhere else.
Does this pass as "it's better than before, let's leave other improvements till later" check?

Reply:
Yeah this makes sense, especially since you're working on a new `wait_for_idle`